test/listen.test.js

```diff
@@ -102,42 +177,111 @@ if (os.platform() !== 'win32') {
   test('listen on socket', t => {
     t.plan(2)
     const fastify = Fastify()
+    t.tearDown(fastify.close.bind(fastify))
     const sockFile = path.join(os.tmpdir(), 'server.sock')
     try {
       fs.unlinkSync(sockFile)
     } catch (e) { }
     fastify.listen(sockFile, (err) => {
       t.error(err)
       t.equal(sockFile, fastify.server.address())
-      fastify.close()
     })
   })
 }
 
 test('listen without callback', t => {
   t.plan(1)
   const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
   fastify.listen(0)
     .then(() => {
       t.is(fastify.server.address().address, '127.0.0.1')
-      fastify.close()
       t.end()
     })
 })
 
+test('listen null without callback with (address)', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+  fastify.listen(null)
+    .then((address) => {
+      t.is(address, 'http://127.0.0.1:' + fastify.server.address().port)
+      t.end()
+    })
+    .catch(err => {
+      t.ok(err)
+    })
+})
+
+test('listen without port without callback with (address)', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+  fastify.listen()
+    .then((address) => {
+      t.is(address, 'http://127.0.0.1:' + fastify.server.address().port)
+      t.end()
```

Comment:
t.end is not necessart if t.plan is used

Reply:
fixed this one